src/inference/src/ie_core.cpp

```diff
@@ -501,12 +514,146 @@ class CoreImpl : public ie::ICore, public std::enable_shared_from_this<ie::ICore
         return res;
     }
 
+    void ApplyAutoBatching(const ie::CNNNetwork& network,
+                           std::string& deviceName,
+                           std::map<std::string, std::string>& config_with_batch) {
+        std::string deviceNameWithBatchSize;
+        if (deviceName.find("BATCH") != std::string::npos) {
+            // explicitly enabled Auto-Batching
+            auto pos = deviceName.find_first_of(":");
+            if (pos != std::string::npos) {
+                deviceNameWithBatchSize = deviceName.substr(pos + 1);
+            }
+        } else {
+            // if the Auto-Batching is disabled explicitly
+            const auto& batch_mode = config_with_batch.find(CONFIG_KEY(ALLOW_AUTO_BATCHING));
+            if (batch_mode != config_with_batch.end()) {
+                const auto disabled = batch_mode->second == CONFIG_VALUE(NO);
+                // no need for this config key in the rest of loading
+                config_with_batch.erase(batch_mode);
+                if (disabled)
+                    return;
+            }
+        }
+        auto device = ov::runtime::parseDeviceNameIntoConfig(deviceName);
+        if (device._deviceName.find("GPU") != std::string::npos) {
```

Comment:
What if user will use not "GPU" device alias?

Reply:
fixed, switched to check the OPTIMAL_BATCHS_SIZE metric support